src/pages/tutor-profile/TutorProfile.jsx

```diff
@@ -41,6 +41,13 @@ const TutorProfile = () => {
     defaultResponse: defaultResponses.array
   })
 
+  useEffect(() => {
+    const tutorVideo = response?.videoLink?.tutor
+    if (!loading && tutorVideo !== null && tutorVideo !== '') {
+      setVideoPreview(false)
+    }
+  }, [loading, response])
+
```

Comment:
`useEffect` and state for storing video preview is redundant, because it is derived state. Please remove `useState` and `useEffect` for this. I suggest refactoring as following:

```suggestion
<VideoPresentation
    video={response?.videoLink?.[userRole]}
    videoPreview={loading || !tutorVideo}
 />
```

Reply:
I don't think it's redundant. 
this implementation is necessary because of the different behavior of mock implication. at 
Tutor profile and student profile 

Also in the code you suggested there is no check for the value of an empty string.
When such data arrives, the video player is displayed incorrectly.

Reply:
I think you can add checks for student into condition, but extra hook usage is not good

Reply:
I understand, but there's a downside. 
a userfect can protect us from race conditions.
Also the realization that we have now is quite readable. 
We can see both the required checks that a string must pass and under what condition it should display the video preview.

Reply:
Can you explain in more details why your solution is better than proposed one? Because I don't get a point why it protects us from race condition

Reply:
I meant it in terms of my code. 
![image](https://github.com/ita-social-projects/SpaceToStudy-Client/assets/101183408/2c892a81-071d-4fb7-bcc4-8f2a9d80d0a2)
if we use this setter (thereby explicitly specifying the state when we should not show video previews)
without using the userfect 
we get this result 
![image](https://github.com/ita-social-projects/SpaceToStudy-Client/assets/101183408/1efae8f2-ec28-49ac-98f0-ff012d203419)

Reply:
You should remove `useState`, it is also redundant here. Thats why you get an error with infinite rerenders

Reply:
![image](https://github.com/ita-social-projects/SpaceToStudy-Client/assets/101183408/1698fa3e-7ae6-4c24-8bfa-9b98265b8428)
fixed

Reply:
I also suggest renaming `showPresentation` to something like `VideoPresentationComponent` to indicate that it is a `jsx` component and not a condition

Reply:
fixed

Reply:
Good job!